Lib/asyncore.py

```diff
@@ -131,33 +131,30 @@ def poll(timeout=0.0, map=None):
             is_r = obj.readable()
             is_w = obj.writable()
             if is_r:
-                r.append(fd)
+                r.append(obj)
             # accepting sockets should not be writable
             if is_w and not obj.accepting:
-                w.append(fd)
+                w.append(obj)
             if is_r or is_w:
-                e.append(fd)
+                e.append(obj)
         if [] == r == w == e:
             time.sleep(timeout)
             return
 
         r, w, e = select.select(r, w, e, timeout)
```

Comment:
Just write a second PR when the first one is merged. I would just be easier to review it. IMHO you are fixing two bugs at once, it makes the review harder.